src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralParsingTests.cs

```diff
@@ -0,0 +1,1870 @@
+    }
+
+    [Fact]
+    public void OuterVerbatimMiddleRawInnerNormal()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@""{$""""""{$""{0}""}""""""}"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterVerbatimMiddleRawInnerVerbatim()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@""{$""""""{@$""{0}""}""""""}"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterVerbatimMiddleRawInnerRaw()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $@""{$""""""{$""""""{0}""""""}""""""}"";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleNormalInnerNormal()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{$""{$""{0}""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleNormalInnerVerbatim()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{$""{$@""{0}""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleNormalInnerRaw()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{$""{$""""""{0}""""""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleVerbatimInnerNormal()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{@$""{$""{0}""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleVerbatimInnerVerbatim()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{@$""{@$""{0}""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleVerbatimInnerRaw()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{@$""{$""""""{0}""""""}""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleRawInnerNormal()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{$""""""{$""{0}""}""""""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleRawInnerVerbatim()
+    {
+        var text = @"
+class C
+{
+    void M()
+    {
+        var v = $""""""{$""""""{@$""{0}""}""""""}"""""";
+    }
+}";
+
+        CreateCompilation(text).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OuterRawMiddleRawInnerRaw()
+    {
+        var text = @"
+class C
+{
+    void M()
```

Comment:
tip: you can save the class and method declaration, using top-level statements.
nit: Also, we recommend to execute tests that don't have errors (see my earlier comment)